GoogleDataTransport/GDTCORLibrary/Public/GDTCORClock.h

```diff
@@ -27,11 +27,17 @@ NS_ASSUME_NONNULL_BEGIN
 /** The offset from UTC in seconds. */
 @property(nonatomic, readonly) int64_t timezoneOffsetSeconds;
 
-/** The kernel boot time when this clock was created. */
-@property(nonatomic, readonly) int64_t kernelBootTime;
+/** The kernel boot time when this clock was created in nanoseconds. */
+@property(nonatomic, readonly) int64_t kernelBootTimeNanoseconds;
 
-/** The device uptime when this clock was created. */
-@property(nonatomic, readonly) int64_t uptime;
+/** The device uptime when this clock was created in nanoseconds. */
+@property(nonatomic, readonly) int64_t uptimeNanoseconds;
+
+@property(nonatomic, readonly) int64_t kernelBootTime DEPRECATED_MSG_ATTRIBUTE(
+    "Please use `kernelBootTimeNanoseconds` instead");
+
+@property(nonatomic, readonly)
+    int64_t uptime DEPRECATED_MSG_ATTRIBUTE("Please use `uptimeNanoseconds` instead");
```

Comment:
Do you want to qualify that this is buggy? This would enable all team who are depending on the Firelog library would know that they have to update to the newer property?

Reply:
The current implementation just forwards the call to the new properties so there is no actual harm in using deprecated ones except they are going to be removed in the next major release.